pkg/provider/infisical/api/api.go

```diff
@@ -218,7 +218,8 @@ func (a *InfisicalClient) GetSecretByKeyV3(data GetSecretByKeyV3Request) (string
 	if err != nil {
 		return "", err
 	}
-	if rawRes.StatusCode == 400 {
+
+	if rawRes.StatusCode == 400 || rawRes.StatusCode == 404 || rawRes.StatusCode == 500 {
```

Comment:
Can you please make this to a check `>400`

We have a good format for error message: https://github.com/Infisical/infisical/blob/main/backend/src/server/plugins/error-handler.ts

So every error will be shown uniformly to user